services/management/dbaas/kubernetes_server.go

```diff
@@ -133,6 +141,40 @@ func (k kubernetesServer) RegisterKubernetesCluster(ctx context.Context, req *db
 		return nil, err
 	}
 
+	settings, err := models.GetSettings(k.db.Querier)
+	if err != nil {
+		return nil, err
+	}
+	if settings.PMMPublicAddress != "" {
```

Comment:
```suggestion
	if settings.PMMPublicAddress == "" {
	      return &dbaasv1beta1.RegisterKubernetesClusterResponse{}, nil
	}
```
(minor reduce indentation)